### ISSUE TITLE
[Proposal]: Use of goupx to reduce the size of binaries

### ISSUE THREAD

**Issue by aledbf** [deis/builder#176]
```
$ goupx rootfs/usr/bin/boot
2016/02/11 22:27:37 {Class:ELFCLASS64 Data:ELFDATA2LSB Version:EV_CURRENT OSABI:ELFOSABI_NONE ABIVersion:0 ByteOrder:LittleEndian Type:ET_EXEC Machine:EM_X86_64 Entry:4580352}
2016/02/11 22:27:37 File fixed!
                       Ultimate Packer for eXecutables
                          Copyright (C) 1996 - 2013
UPX 3.91        Markus Oberhumer, Laszlo Molnar & John Reiser   Sep 30th 2013

        File size         Ratio      Format      Name
   --------------------   ------   -----------   -----------
  14844160 ->   4119800   27.75%  linux/ElfAMD   boot

Packed 1 file.
```

**Comment by bacongobbler** [comment]
related: https://github.com/golang/go/issues/6853

**Comment by arschles** [comment]
I'm all for this. @aledbf do the basics (`git push`, server startup, ...)  still work after the binary is packed?

**Comment by aledbf** [comment]
@arschles yes (I'm already using this change)
```
deis/builder    git-11869e8            2e28cbdc884d        23 seconds ago       40.48 MB
<none>          <none>                 90153bc0ae4e        About a minute ago   51.62 MB
```

**Comment by arschles** [comment]
@aledbf ok, can you submit a PR for the change so that others can test as well?

**Comment by mboersma** [comment]
I love this proposal. I've seen `upx` routinely shrink Go binaries by 80% or more, which can have an obvious impact on Docker image sizes.

I have heard anecdotally that `upx`-ed binaries sometimes trigger virus scanner false positives, however. I've never seen this myself but let's keep an eye out. :eyes:

**Comment by mboersma** [comment]
Closed by #179.
